Library/OcMainLib/OpenCoreMisc.c

```diff
@@ -470,8 +472,35 @@ OcMiscEarlyInit (
     gBS->SetWatchdogTimer (0, 0, 0, NULL);
   }
 
-  if (Config->Misc.Debug.SerialInit) {
-    SerialPortInitialize ();
+  if (Config->Misc.Serial.Init) {
+    //
+    // Validate the size of PciDeviceInfo. Abort on error.
+    //
+    PciDeviceInfoSize = Config->Misc.Serial.PciDeviceInfo.Size;
+    if (PciDeviceInfoSize > OC_SERIAL_PCI_DEVICE_INFO_MAX_SIZE) {
+      DEBUG ((DEBUG_INFO, "OC: Aborting serial port init with borked PciDeviceInfo size %u\n", PciDeviceInfoSize));
```

Comment:
imo return here to match how we do error-handling in other places

Reply:
Why? This is not fatal, and we have other tasks below.

Reply:
Ah, the diff was cropped badly, didn't see